core/src/main/java/com/novoda/noplayer/internal/exoplayer/mediasource/MediaSourceFactory.java

```diff
@@ -64,9 +64,12 @@ private MediaSource createH264MediaSource(DefaultDataSourceFactory defaultDataSo
     }
 
     private MediaSource createDashMediaSource(DefaultDataSourceFactory defaultDataSourceFactory,
-                                              Uri uri) {
+                                              Uri uri,
+                                              MediaSourceEventListener mediaSourceEventListener) {
         DefaultDashChunkSource.Factory chunkSourceFactory = new DefaultDashChunkSource.Factory(defaultDataSourceFactory);
         DashMediaSource.Factory factory = new DashMediaSource.Factory(chunkSourceFactory, defaultDataSourceFactory);
-        return factory.createMediaSource(uri);
```

Comment:
I wonder if this is a smell that we're applying the eventListeners at the wrong level?